loama/src/components/IconButton.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+    <button :class="type">
+        <component :v-if="leftIcon" :is="leftIcon" :size="24"/>
+        <span><slot></slot></span>
+        <component :v-if="rightIcon" :is="rightIcon" :size="24"/>
+    </button>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue';
+
+const props = withDefaults(defineProps<{
+    /**
+     * The icon to display on the left side of the button. Should be a Phosphor icon from the `@phosphor-icons/vue` package.
+     */
+    leftIcon?: Object,
```

Comment:
wouldn't it make more sense to take it as a `string` representing the icon name instead of sending a component as prop?

Reply:
nvm just saw below that its straight imports its not something like `<icon name="" />`

Reply:
Yeah and I tried finding a type of like Vue Component, like you can do in React *(E.g. `ReactNode`, etc...)*, but haven't found anything like that for Vue atm